src/domain/product.ts

```diff
@@ -7,21 +7,18 @@ export type Product = {
   categoryId: number;
 };
 
+export type ProductRequest = Omit<Product, "id">;
+
 export function fetchData() {
-  return fetch(`http://localhost:3000/product`)
-    .then((result) => {
-      return result.json();
-    })
-    .then((data: Product[]) => {
-      return [...data];
-    });
+  return fetch(`http://localhost:3000/product`).then((result) => {
+    return result.json();
+  });
```

Comment:
Ini typing `Product[]` bisa ditaruh sini

```suggestion
  return fetch(`http://localhost:3000/product`)
    .then((result) => result.json())
    .then((data: Product[]) => data);
```

Reply:
updated mas